[WIP] developer guide (#3531)
knative/docs: docs/getting-started/_index.md

---
title: "Getting started"
weight: 01
type: "docs"
showlandingtoc: "false"
---

## Installing Knative

To get started with a development Knative installation, you can install Knative components and Kubernetes on a local Docker Daemon by using _Knative on [`kind`](https://kind.sigs.k8s.io/docs/user/quick-start)_ (`konk`).

## Prerequisites

- Install [`kind`](https://kind.sigs.k8s.io/docs/user/quick-start) before you use `konk`.

## About `konk`

`konk` is a shell script that:

> <!--does konk install kind? Confirm-->

Reviewer: ```suggestion
```
It does not, kind ought to be a pre-requisite

1. Runs a script called [`01-kind.sh`](https://github.com/csantanapr/knative-kind/blob/master/01-kind.sh), that checks if `kind` is installed. If `kind` is installed, the script creates a Kubernetes cluster on `kind` called `knative`.
1. Runs a script called [`02-serving.sh`](https://github.com/csantanapr/knative-kind/blob/master/02-serving.sh), that:
    - Installs Knative Serving with Kourier as the default networking layer.
    - Installs nip.io as the DNS, and provides a default port-forwarding configuration for the `knative` cluster.
1. Runs a script called [`04-eventing.sh`](https://github.com/csantanapr/knative-kind/blob/master/04-eventing.sh), that installs Knative Eventing, and creates a default InMemoryChannel resource and channel-based broker on the `knative` cluster.
<!-- TODO: Add links for serving and kourier/networking docs sections-->
<!--TODO: Add links for channel and broker resources and Knative Eventing docs-->

## Installing Knative using `konk`

To install `konk`, enter the following command:

```
curl -sL install.konk.dev | bash
```

> 

Reviewer: ```suggestion

**NOTE:** `konk` is currently a beta product, and we are looking for feedback (positive, constructive or otherwise. Please open an issue in https://github.com/knative/ux to let us know about your experience! We review every issue. 
```

## Install CLI tools

You can interact with your Knative on your cluster by using the `kubectl` and `kn` CLI tools.

- `kubectl` is used primarily to complete Kubernetes cluster administrator tasks.
<!-- TODO: Add link to admin guide-->

> - `kn` is used primarily by developers to create Knative components and set up workflows.

Reviewer: ```suggestion
- `kn` is used primarily by developers to simplify the creation of Knative components and more easily set up workflows.
```

High-lighting benefits of kn
